explorer/interpreter/interpreter.cpp

```diff
@@ -40,6 +40,33 @@ using llvm::isa;
 
 namespace Carbon {
 
+// TODO: Refactor this method & maybe move it to somewhere else
+auto format_action_str(Action& act) -> std::string {
+  std::string str;
+  llvm::raw_string_ostream stream(str);
+  stream << act;
+  auto string_ref = llvm::StringRef(str);
+  llvm::SmallVector<llvm::StringRef> lines;
+  string_ref.split(lines, "\n");
+  auto step = lines.pop_back_val();
+  std::string res;
+
+  res += "\n```\n";
+  // Adds the first line
+  if (!lines.empty()) {
+    res += lines.front();
+    res += lines.size() > 1 ? "\n ...\n" : "\n";
+  }
+
+  // Adds the last line
+  if (lines.size() > 1) {
+    res += lines.back().str() + "\n";
+  }
+  res += "\n```" + step.str();
```

Comment:
Putting the last line outside the \`\`\` doesn't seem to be working very reliably. In the trace output below are things like:

```
// CHECK:STDOUT: ```
// CHECK:STDOUT:
// CHECK:STDOUT: ```i32 .0. --->{{ }}
```

and 

```
// CHECK:STDOUT: ```
// CHECK:STDOUT:
// CHECK:STDOUT: ```Main() .0. --->{{ }}
```

Maybe it'd be better to move this code to render a truncated snippet within \`\`\`s into the action printing logic, and wrap it around the printing of the declaration / expression.

Also, when printing a `DeclarationAction`, you could change the `Action::Print` function to use `Declaration::PrintID` instead of `Declaration::Print`, so we don't print out the body in the first place.

Reply:
> Putting the last line outside the ``` doesn't seem to be working very reliably. In the trace output below are things like:

I'm sorry I missed that, thank you so much for pointing it out!
I'll do the suggested changes in the next commit.